### PR TITLE
[Please Do Not Merge] Added spec for Policy API

### PR DESCRIPTION
**Pull request by asarkar84**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by asarkar84** [comment]
For our next GTM demo in PROD, we need to enable the ARM endpoint for Policy API. Before making ARM PROD manifest update, I am sending this PR to get your opinion and ARM Signoff: 
https://github.com/Azure/azure-rest-api-specs/pull/4241

This is extension to the existing ARM approved PR: #4005. I have taken all files from the this PR and added only Policy API related specs.
Please review the below path only which is added newly in the PR and share your opinion: 
"/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/billingProfiles/{billingProfileId}/policy/latest"

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json
Comment (jhendrixMSFT) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json]: This name really should be `Policy_GetByBillingProfile`.  With the current naming this operation will have a separate SDK client from `Policy_Update` making it clumsy to work with.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json
Comment (KrisBash) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json]: To adhere to convention, this should be "policies" instead of "policy." Also, the convention for singleton resources is to use /default (as in {}/policy/default). Is there a specific reason to use /latest? Are policies other than the latest enumerable?

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json
Comment (KrisBash) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json]: Will there ever be a tertiary value for this (in addition to true and false)? Using a string enum type with allowed values of "true" and "false" allows for the addition of future values without being a breaking change (like changing from bool to string).

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json
Comment (KrisBash) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json]: See comments above on the boolean

**Review by KrisBash — CHANGES_REQUESTED** [review]
Please see my comments

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json
Comment (asarkar84) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json]: Sure, changed from latest to default. 
There was some previous ARM feedback to use latest so thought of using it. That scenario is little different though.

Comment (asarkar84) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json]: As of now policies are always latest.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json
Comment (asarkar84) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json]: There will not be any value other than true or false.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json
Comment (asarkar84) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json]: There will not be any value other than true or false.

**Comment by asarkar84** [comment]
Review comments have been incorporated. Please take a look.

**Review by KrisBash — APPROVED** [review]
Signing off from ARM

**Comment by jhendrixMSFT** [comment]
@asarkar84 the two new examples have some errors, can you please take a look at the [log](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/442493822) and fix?  See the very bottom of the log/ search by the example names `PolicyByBillingProfile` and `UpdateBillingProfile`.  Once these have been resolved should be good to merge.

**Comment by jhendrixMSFT** [comment]
@asarkar84 is this one still on hold?

**Comment by jhendrixMSFT** [comment]
Ping @asarkar84

**Comment by jhendrixMSFT** [comment]
@asarkar84 is this still waiting?

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]
Please take a look at the model validation failures [here](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/443484503).  These will need to be fixed.

**Comment by asarkar84** [comment]
Thank you for the feedback. Closing this PR since this needs to be merged with the larger change for GTM feature. Will send another PR with all the changes.

**Pull request closed without merging**
